service/worker/pernamespaceworker_test.go

```diff
@@ -192,8 +192,23 @@ func (s *perNsWorkerManagerSuite) TestEnabled() {
 	cli1.EXPECT().Close()
 }
 
+/*
+		Given machine has ownership of the namespace
+		When name space change reported
+		Then worker should be started with configuration proportional to administratively set configuration ensuring
+	    fair distribution across all the machines owning the namespace
+*/
 func (s *perNsWorkerManagerSuite) TestMultiplicity() {
-	ns := testns("ns3", enumspb.NAMESPACE_STATE_REGISTERED) // three workers
+	desiredWorkersNumber := 6
+	machinesOwningNamespace := []membership.HostInfo{
+		membership.NewHostInfoFromAddress("other-1"),
+		membership.NewHostInfoFromAddress("other-2"),
+		membership.NewHostInfoFromAddress("self"),
+	}
+	expectedMultiplicity := 2
+
+	ns := testns("ns3", enumspb.NAMESPACE_STATE_REGISTERED)
+	s.manager.config.PerNamespaceWorkerCount = func(_ string) int { return desiredWorkersNumber }
```

Comment:
don't override this here, these are configured here: https://github.com/temporalio/temporal/blob/main/service/worker/pernamespaceworker_test.go#L87-L89
(or you could go the other way and move all the Config overrides to local to each test. I guess that's nicer in some ways.)